infra/build_specified_commit.py

```diff
@@ -54,42 +59,116 @@ def build_fuzzer_from_commit(project_name,
       sanitizer=sanitizer,
       architecture=architecture,
       env_to_add=None,
-      source_path=old_repo_manager.repo_dir)
+      source_path=old_repo_manager.repo_dir,
+      mount_location=os.path.join('/src', old_repo_manager.repo_name))
+
+
+def run_command_in_image(image_name, command):
+  """Runs a specific command inside a docker image and returns the results.
+
+  Args:
+    image_name: The docker image for the command to be run in
+    command: The command to be run in the image
+
+  Returns:
+    the output of the command
+
+  Raises:
+    DockerExecutionError: on commands execution failing
+  """
+  command_to_run = [
+      'docker', 'run', '--rm', '-i', '--privileged', '-t', image_name
+  ]
+  command_to_run.extend(command)
+  print('Running command: %s' % command_to_run)
+  process = subprocess.Popen(command_to_run, stdout=subprocess.PIPE)
+  out, err = process.communicate()
+  if err:
+    raise DockerExecutionError('Error running command: %s with error: %s' %
+                               (command_to_run, err.decode('ascii')))
+  if out:
+    return out.decode('ascii'), process.returncode
+  return None, process.returncode
+
+
+def check_docker_for_commit(docker_image_name, dir_name, example_commit):
+  """ Checks a docker image directory for a specific commit.
+
+  Args:
+    docker_image_name: The name of the projects docker image
+    dir_name: The name of the directory to test for the commit
+    example_commit: The commit SHA to check for
 
+  Returns:
+    True if docker image directory contains that commit
+  """
+  dir_to_check = '/src/' + dir_name
+
+  #Check if valid git repo
+  out, returncode = run_command_in_image(
+      docker_image_name, ['/bin/bash', '-c', 'ls -a ' + dir_to_check])
+  if '.git' not in out:
+    return False
+
+  #Check if history fetch is needed
+  check_shallow_command = '[ -f ' + dir_to_check + '/.git/shallow ]'
+  out, returncode = run_command_in_image(
+      docker_image_name, ['/bin/bash', '-c', check_shallow_command])
+
+  if returncode == 0:
+    #Check if commit exists
+    _, returncode = run_command_in_image(docker_image_name, [
+        '/bin/bash', '-c', 'cd ' + dir_to_check +
+        '; git fetch --unshallow; git cat-file -e ' + example_commit
```

Comment:
You probably want to use ` && ` instead of `; `.
ampersands will prevent the latter commands from executing if the first fails.